sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -297,46 +277,83 @@ public boolean isRetryAfter(final @NotNull String itemType) {
     int responseCode = -1;
 
     try (final OutputStream outputStream = connection.getOutputStream();
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, UTF_8))) {
+        final GZIPOutputStream gzip = new GZIPOutputStream(outputStream);
+        final Writer writer = new BufferedWriter(new OutputStreamWriter(gzip, UTF_8))) {
       serializer.serialize(envelope, writer);
 
-      // need to also close the input stream of the connection
-      connection.getInputStream().close();
       options.getLogger().log(DEBUG, "Envelope sent successfully.");
       return TransportResult.success();
     } catch (IOException e) {
-      try {
-        responseCode = connection.getResponseCode();
+      final TransportResult errorResult = getResultFromErrorCode(connection, e);
+      responseCode = errorResult.getResponseCode();
+      return errorResult;
+    } catch (Exception e) {
+      return getDefaultErrorResult(e);
+    } finally {
+      updateRetryAfterLimits(connection, responseCode);
 
-        // TODO: 403 part of the protocol?
-        if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
-          options
-              .getLogger()
-              .log(DEBUG, "Envelope was rejected by the Sentry server due to a filter.");
-        }
-        logErrorInPayload(connection);
-        return TransportResult.error(responseCode);
-      } catch (IOException responseCodeException) {
-        // this should not stop us from continuing. We'll just use -1 as response code.
-        options
-            .getLogger()
-            .log(WARNING, "Failed to obtain response code while analyzing event send failure.", e);
-      }
+      closeAndDisconnectConnection(connection);
+    }
+  }
+
+  /**
+   * Closes the Response stream and disconnect the connection
+   *
+   * @param connection the HttpURLConnection
+   */
+  private void closeAndDisconnectConnection(final @NotNull HttpURLConnection connection) {
```

Comment:
`l`: Maybe we can remove the connection in the name and rename it to `closeAndDisconnect`, because calling the function looks like this `closeAndDisconnectConnection(connection)` and there connection is twice. That's some unnecessary noise for me.

Reply:
yes, makes sense